DataDog/datadog-operator
# Add cilium network policies

### What does this PR do?

Add cilium network policy manager as a follow up to implementing `ApplyGlobalSettings()`

Note: does not include feature-specific policies

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

* Install Cilium in a cluster: [kind cluster example using cilium v1.12](https://docs.cilium.io/en/v1.12/gettingstarted/kind/)
* Spin up operator + DatadogAgent and check that cilium network policies are created
```yaml
spec:
  global:
    networkPolicy:
      create: true
      flavor: cilium
```
```
> kubectl get ciliumnetworkpolicy   
NAME                            AGE
datadog-agent                   14s
datadog-cluster-agent           14s
datadog-cluster-checks-runner   14s
```
